lib/ddtrace/configuration/settings.rb

```diff
@@ -277,23 +427,45 @@ def runtime_metrics(options = nil)
         o.lazy
       end
 
+      # [Continuous Integration Visibility](https://docs.datadoghq.com/continuous_integration/) configuration.
+      # @public_api
       settings :test_mode do
+        # Enable test mode. This allows the tracer to collect spans from test runs.
+        #
+        # It also prevents the tracer from collecting spans in a production environment. Only use in a test environment.
+        #
+        # @default `DD_TRACE_TEST_MODE_ENABLED` environment variable, otherwise `false`
+        # @return [Boolean]
         option :enabled do |o|
           o.default { env_to_bool(Ext::Test::ENV_MODE_ENABLED, false) }
           o.lazy
         end
 
+        # TODO: Remove this configuration.
+        # TODO: It is not necessary, as it be configured by the default flushing trace configuration.
         option :trace_flush do |o|
           o.default { nil }
           o.lazy
         end
 
+        # TODO: Remove this configuration.
+        # TODO: It is not necessary, as it be configured by the default writer trace configuration.
         option :writer_options do |o|
```

Comment:
Are these TODOs going to be visible to customers? If so, should we mark them as deprecated instead?

Reply:
YARD is naive, it keeps everything that is a comment before the declaration, TODOs and all.
I'll see if I can clean them up.

Reply:
TODOs and DEVs don't surface anymore.